src/decoder/mod.rs

```diff
@@ -177,8 +177,10 @@ impl<R: Read> Decoder<R> {
     /// Most image metadata will not be read until `read_info` is called, so those fields will be
     /// None or empty.
     pub fn read_header_info(&mut self) -> Result<&Info, DecodingError> {
+        let mut buf = Vec::new();
         while self.read_decoder.info().is_none() {
-            if self.read_decoder.decode_next(&mut Vec::new())?.is_none() {
+            buf.clear();
+            if self.read_decoder.decode_next(&mut buf)?.is_none() {
```

Comment:
There should probably be a comment/assertion here, but no data is actually ever pushed into the Vec because the loop stops as soon as the image header gets read